static/test/javascripts/spec/common/onward/breaking-news.spec.js

```diff
@@ -0,0 +1,286 @@
+            knownAlertIDsStorageKey = 'gu.breaking-news.hidden',
+            server;
+
+        function alertThatIs(type, options) {
+            options = defaults(options || {}, {
+                collection: 'uk',
+                age: 1
+            });
+            var sAge = options.age !== 1 ? options.age + 'min ' : '';
+            return {
+                href: options.collection,
+                content: [{
+                    headline: sAge + options.collection + ' ' + type + ' headline',
+                    trailText: options.collection + ' ' + type + ' trailText',
+                    id: options.collection + '_' + type,
+                    frontPublicationDate: Date.now() - (1000 * 60 * options.age)
+                }]
+            };
+        }
+
+        function mockBreakingNewsWith(collections) {
+            return new Promise(function (resolve, reject) {
+                injector.mock({
+                    'common/utils/storage': storage,
+                    'common/utils/ajax': ajax
+                }).require(['common/modules/onward/breaking-news'], function (breakingNews) {
+                    server = sinon.fakeServer.create();
+                    server.respondImmediately = true;
+                    server.autoRespond = true;
+                    server.respondWith(breakingNewsURL, [200, {'Content-Type': 'application/json'}, JSON.stringify({
+                        webTitle: 'Breaking News',
+                        collections: collections
+                    })]);
+                    breakingNews().then(function(result) {
+                        // make sure the DOM has finished updating
+                        requestAnimationFrame(function () {
+                            resolve(result);
+                        });
+                    }, reject);
+                    server.restore();
+                }, function (e) {
+                    console.log(e);
+                });
+            });
+        }
+
+        beforeAll(function () {
+            ajax = jasmine.createSpy().and.callFake(ajax);
+            config.page.edition = 'UK';
+        });
+
+        beforeEach(function (done) {
+            $('body').html('<div class="js-breaking-news-placeholder breaking-news breaking-news--hidden breaking-news--fade-in" data-link-name="breaking news" data-component="breaking-news"></div>');
+            requestAnimationFrame(done);
+        });
+
+        describe('user cannot dismiss alerts', function () {
+            beforeEach(function () {
+                sinon.stub(storage.local, 'isAvailable').returns(false);
+            });
+
+            it('should not try and fetch the json', function (done) {
+                mockBreakingNewsWith([]).then(function () {
+                    fail('user cannot use local storage, but we seem to think things are okish');
+                    done();
```

Comment:
The best way (IMO) to do this is Jasmine (which unfortunately doesn't support promises like mocha or others) is

```js
mockBreakingNewsWith([]).then(() => {
   done.fail('message')
}, () => {
   // some expectations
})
.then(done)
.catch(done.fail)
```

It's easier when you're testing for resolved promises

```js
returnsAPromise()
.then(() => {  /* expectations */ })
.then(done)
.catch(done.fail)
```